src/components/NavBarButton.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button, ButtonProps } from "react-aria-components";
+import styled from "styled-components";
+
+export type NavBarButtonProps = {
+  label?: string;
+  icon?: string | React.ReactNode;
+  className?: string;
+  "aria-label"?: string;
+} & ({ label: string } | { "aria-label": string | undefined });
```

Comment:
`{ "aria-label": string | undefined }` should be covered by `"aria-label"?: string;` which makes the whole `OR` statement redundant

Reply:
I guess the one exception is `"aria-label"?: string;` allows aria-label to not be set, while `"aria-label": string | undefined` requires it to be explicitly set to undefined... that seems like such an odd distinction to make though?

Reply:
This was to satisfy TS while maintaining full test coverage without using `any` in the spec but in hindsight was over-complicated, spreading the props took it out of the equation.